src/edu/csus/ecs/pc2/core/execute/Executable.java

```diff
@@ -2524,10 +2525,33 @@ public String substituteAllStrings(Run inRun, String origString) {
                 newString = replaceString(newString, "{:exitvalue}", Integer.toString(executionData.getExecuteExitValue()));
                 newString = replaceString(newString, "{:executetime}", Long.toString(executionData.getExecuteTimeMS()));
             }
+            
             String pc2home = new VersionInfo().locateHome();
             if (pc2home != null && pc2home.length() > 0) {
                 newString = replaceString(newString, "{:pc2home}", pc2home);
             }
+            
+            // TODO REFACTOR replace vars with constants for: memlimit, sandboxcommandline,sandboxprogramname
+            newString = replaceString(newString, "{:memlimit}", Integer.toString(problem.getMemoryLimitMB()));
+
+            if (newString.indexOf("{:sandboxcommandline}") > -1) {
+                if (!StringUtilities.isEmpty(problem.getSandboxCmdLine())) {
+
+                    newString = replaceString(newString, "{:sandboxcommandline}", problem.getSandboxCmdLine());
+                    newString = substituteAllStrings(inRun, newString);
+                } else {
+                    // remove {:sandbox} if no sandbox defined in pc2
+                    newString = replaceString(newString, "{:sandboxcommandline}", "");
```

Comment:
I don't understand this statement.  The comment above it says it is removing "{:sandbox}", but the statement itself is replacing "{:sandboxcommandline]".  Either the comment is wrong or the statement is wrong (or I REALLY don't understand what is going on).  Either the comment needs to be corrected or the statement needs to be corrected.  (Or I would accept a brief explanation of how the statement actually does do what the comment says it does...)

Reply:
typo in comment.

The condition is: if the variable sandboxcommandline is empty then substitute with ""
if not empty use the value from problem.

Reply:
So, are you intending to correct it?   (As someone famously once said, "If the documentation doesn't match the code, both are wrong"... ;)

Reply:
pushing fix in comment.

Reply:
Well, I guess totally eliminating documentation is one way to make sure the documentation isn't inaccurate :(